app/fileManager.js

```diff
@@ -59,17 +69,61 @@ export default function(state, emitter) {
   function render() {
     emitter.emit('render');
   }
+  /*
+
+  function updateDloads() {
+    emitter.emit('updateDloads');
+  }
+*/
+
+  const dlimitArray = [];
+  const fileIdArray = [];
+  function ifRerender(file) {
+    if (fileIdArray.indexOf(file.id) === -1) {
+      const fileDetails = {
+        fileId: file.id,
+        dlimit: file.dlimit,
+        dtotal: file.dtotal
+      };
+      dlimitArray.push(fileDetails);
+      return false;
+    } else {
+      for (const fileDetail of dlimitArray) {
+        if (file.id === fileDetail.fileId) {
+          if (
+            file.dlimit !== fileDetail.dlimit ||
+            file.dtotal !== fileDetail.dtotal
+          ) {
+            fileDetail.dtotal = file.dtotal;
+            fileDetail.dlimit = file.dlimit;
+            return true;
+          } else {
+            return false;
+          }
+        }
+      }
+    }
+  }
 
   async function checkFiles() {
     const files = state.storage.files;
     let rerender = false;
+
     for (const file of files) {
-      const ok = await exists(file.id);
-      if (!ok) {
+      const receivedFile = await getDLCounts(file).then(function(data) {
+        return data;
+      });
+      console.log(receivedFile);
```

Comment:
This has been the main problem. The `console.log()` statement here prints an undefined statement because of which no further execution happens. 

There is something wrong with the promise returned here as the `.then` function receives no data here. Please help .